### ISSUE TITLE
Question: Combining Minkowski Engine with standard pytorch layers.

### ISSUE THREAD

**Issue by TommasoBendinelli** [NVIDIA/MinkowskiEngine#219]
I would like to know if it is possible to use layers from your package with standard pytorch layers.
Particularly I would like my point cloud dataset and then pass the result to the rest of my network that I have already implement in standard pytorch code.

**Comment by xmyqsh** [comment]
Certainly!
Just add the following line between them.
```
dense_tensor = sparse_tensor.dense(min_coords, max_coords);
```

**Comment by hillegass** [comment]
I am doing something similar: I have a lidar point cloud, and I'm making a MinkowskiEngine sparse tensor. Then I have a module that puts it through a few MInkowskiEngine convolutions, and then converting it to a dense tensor, doing a few transposes, and a reshape:
```
class MinkConv(ME.MinkowskiNetwork):
...
    def forward(self, x):
        # MinkowskiEngine sparse convolutions
        out = self.conv1(x)
        out = self.conv2(out)
        out = self.conv3(out)
        out = self.conv4(out)
        out = self.conv5(out)
        # Non-standard stuff
        (dense, a, b) = out.dense(min_coords=self.out_min, max_coords=self.out_max)
        dense = torch.transpose(dense, 3, 4)
        dense = torch.transpose(dense, 2, 3)
        dense = dense.reshape(1, 64, 512, 512)
        return dense
```

Is auto-differentiation going to just work? Or do I need to do something special (like register_backward_hook ) before this module will do gradient descent?

**Comment by chrischoy** [comment]
It works.

Check `.grad` to confim.
